build_runner/lib/src/package_graph/apply_builders.dart

```diff
@@ -61,13 +61,32 @@ BuilderApplication applyToRoot(Builder builder,
 /// read by a later builder, or is used as a primary input to a later builder.
 /// If no build actions read the output of an optional action, then it will
 /// never run.
+///
+/// Any existing Builders which match a key in [appliesBuilders] will
+/// automatically be applied to any target which runs this Builder, whether
+/// because it matches [filter] or because it was enabled manually.
 BuilderApplication apply(String builderKey,
-        List<BuilderFactory> builderFactories, PackageFilter filter,
-        {bool isOptional, bool hideOutput, InputSet defaultGenerateFor}) =>
-    new BuilderApplication._(builderKey, builderFactories, filter,
-        isOptional: isOptional,
-        hideOutput: hideOutput,
-        defaultGenerateFor: defaultGenerateFor);
+    List<BuilderFactory> builderFactories, PackageFilter filter,
+    {bool isOptional,
+    bool hideOutput,
+    InputSet defaultGenerateFor,
+    Iterable<String> appliesBuilders}) {
+  if (appliesBuilders != null &&
```

Comment:
Not sure this error checking is really necessary? What situation is it trying to handle? (I don't see an empty applies_builders list as really needing anything special)